[CI] TransportHandshakerTests testHandshakeRequestFutureVersionsCompatibility failing

**Build Scans:**
- [elasticsearch-intake #37803 / part1](https://gradle-enterprise.elastic.co/s/wrb2zhoe6qw7a)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.transport.TransportHandshakerTests.testHandshakeRequestFutureVersionsCompatibility" -Dtests.seed=74B88EF9EAC0771D -Dtests.locale=en-IN -Dtests.timezone=Asia/Qostanay -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.transport.TransportHandshakerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testHandshakeRequestFutureVersionsCompatibility'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [TEST-SynonymsManagementAPIServiceTests.testPutSynonymRuleUpgradeGuardRejectsOnOldCluster-seed#[74B88EF9EAC0771D]][testPutSynonymRuleUpgradeGuardRejectsOnOldCluster]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:104)
	org.elasticsearch.search.SearchResponseUtils$SearchResponseBuilder.build(SearchResponseUtils.java:229)
	org.elasticsearch.search.SearchResponseUtils.successfulResponse(SearchResponseUtils.java:91)
	org.elasticsearch.synonyms.SynonymsManagementAPIServiceTests$SynonymSetExistsAndCountClient.doExecute(SynonymsManagementAPIServiceTests.java:243)
	org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:160)
	org.elasticsearch.client.internal.FilterClient.doExecute(FilterClient.java:57)
	org.elasticsearch.client.internal.OriginSettingClient.doExecu
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testHandshakeRequestFutureVersionsCompatibility
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -334,60 +334,21 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexFailsWhenPitRelocationFails
   issue: https://github.com/elastic/elasticsearch/issues/146650
-- class: org.elasticsearch.transport.ClusterConnectionManagerTests
-  method: testConcurrentConnectsAndDisconnects
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.RemoteClusterClientTests
-  method: testQuicklySkipUnavailableClusters
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testHandshakeRequestFutureVersionsCompatibility
-  issue: https://github.com/elastic/elasticsearch/issues/146743
 - class: org.elasticsearch.simdvec.ESVectorUtilTests
   method: testPow2DiffAndScale
   issue: https://github.com/elastic/elasticsearch/issues/146690
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testHandshakeRequestAndResponse
-  issue: https://github.com/elastic/elasticsearch/issues/146743
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/146553
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexTaskRelocatesOnNodeShutdown
   issue: https://github.com/elastic/elasticsearch/issues/146696
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testHandshakeError
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.RemoteClusterClientTests
-  method: testEnsureWeReconnect
-  issue: https://github.com/elastic/elasticsearch/issues/146743
 - class: org.elasticsearch.simdvec.ESVectorUtilTests
   method: testLogSumExpDiff
   issue: https://github.com/elastic/elasticsearch/issues/146700
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testHandshakeResponseFromOlderNodeWithPatchedProtocol
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testHandshakeResponseFromOlderNode
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.ClusterConnectionManagerTests
-  method: testConnectAfterClose
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.ClusterConnectionManagerTests
-  method: testConnectFailsDuringValidation
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testIncompatibleHandshakeRequest
-  issue: https://github.com/elastic/elasticsearch/issues/146743
-- class: org.elasticsearch.transport.TransportHandshakerTests
-  method: testReadV8HandshakeRequest
-  issue: https://github.com/elastic/elasticsearch/issues/146743
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeForkIT
   method: test {csv-spec:stats_sparkline.sparklineComplexWithGrouping}
   issue: https://github.com/elastic/elasticsearch/issues/146729
-- class: org.elasticsearch.transport.ClusterConnectionManagerTests
-  method: testConcurrentConnectsDuringClose
-  issue: https://github.com/elastic/elasticsearch/issues/146743
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeApproximationIT
   method: test {csv-spec:inlinestats.pushDown_OneGroupingFilter_PastInlineJoinWithInnerFilterAndOuterFilter}
   issue: https://github.com/elastic/elasticsearch/issues/146737

